packages/core/contracts/bot-helpers/atomic-disputer/ReserveCurrencyDisputer.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "@uniswap/lib/contracts/libraries/TransferHelper.sol";
+import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router01.sol";
+
+import "../../common/implementation/FixedPoint.sol";
+
+/**
+ * @title ReserveCurrencyDisputer
+ * @notice Helper contract to enable a disputer to hold one reserver currency and dispute against any number of
+ * financial contracts. Is assumed to be called by a DSProxy which holds reserve currency.
+ */
+
+contract ReserveCurrencyDisputer {
+    using SafeMath for uint256;
+    using FixedPoint for FixedPoint.Unsigned;
+
+    /**
+     * @notice Swaps required amount of reserve currency to collateral currency which is then used to dispute a liquidation.
+     * @dev Any collateral the contract has will be used before anything is purchased on Uniswap.
+     * @param uniswapRouter address of the uniswap router used to facilitate trades.
+     * @param financialContract address of the financial contract on which the liquidation is occurring.
+     * @param reserveCurrency address of the token to swap for collateral. THis is the common currency held by the DSProxy.
+     * @param sponsor address of the sponsor who's liquidation is disputed.
+     * @param liquidationId index of the liquidation for the given sponsor.
+     * @param maxReserveTokenSpent maximum number of reserve tokens to spend in the trade. Bounds slippage.
+     * @param deadline abort the trade and dispute if the transaction is mined after this timestamp.
+     **/
+    function swapDispute(
+        address uniswapRouter,
+        address financialContract,
+        address reserveCurrency,
+        uint256 liquidationId,
+        address sponsor,
+        FixedPoint.Unsigned calldata maxReserveTokenSpent,
```

Comment:
nit: why pass this in as a fixed point since it seems like we only use it when passing its rawValue into functions that take uint256? Put another way, we don't use any fixed point operations, so it might be simpler just to pass it in as a uint256. Thoughts?

Reply:
good point. have changed it.